tests/e2e/browse-bills.spec.ts

```diff
@@ -0,0 +1,637 @@
+      "energy",
+      "housing",
+      "security"
+    ]
+    return words[Math.floor(Math.random() * words.length)]
+  }
+
+  /**
+   * Function to perform a search.
+   * @param page - The Playwright page object.
+   * @param searchTerm - The search term to use.
+   */
+  const performSearch = async (page: Page, searchTerm: string) => {
+    await page.fill('input[placeholder="Search For Bills"]', searchTerm)
+    await page.keyboard.press("Enter")
+  }
+
+  /**
+   * Function to wait for search results to change.
+   * @param page - The Playwright page object.
+   * @param initialResultCount - The initial result count to compare against.
+   */
+  const waitForResultsToChange = async (
+    page: Page,
+    initialResultCount: string
+  ) => {
+    await page.waitForFunction(initialResultCount => {
+      const currentResultCount = document.querySelector(
+        ".ResultCount__ResultContainer-sc-3931e200-0"
+      )?.textContent
+      return currentResultCount !== initialResultCount
+    }, initialResultCount)
+  }
+
+  /**
+   * Function to get category labels text content.
+   * @param page - The Playwright page object.
+   * @returns An array of text content of category labels.
+   */
+  const getCategoryLabelsTextContent = async (
+    page: Page
+  ): Promise<string[]> => {
+    const categoryLabels = page.locator(".ais-CurrentRefinements-categoryLabel")
+    await categoryLabels.first().waitFor({ state: "visible" })
+
+    return categoryLabels.evaluateAll(labels =>
+      labels.map(label => label.textContent || "")
+    )
+  }
+
+  /**
+   * Function to check the full content of the first bill on the page.
+   * @param page - The Playwright page object.
+   * @param searchTerm - The search term to validate in the bill content.
+   */
+  const checkFirstBill = async (page: Page, searchTerm: string) => {
+    const firstBillLink = await page.$eval(
+      "li.ais-Hits-item a",
+      link => (link as HTMLAnchorElement).href
+    )
+
+    await page.goto(firstBillLink)
+
+    const readmorebtn = page.locator(".Summary__StyledButton-sc-791f19-3")
+    await readmorebtn.click()
+
+    const fullContent = await page.textContent(
+      ".Summary__FormattedBillDetails-sc-791f19-4"
+    )
+
+    if (
+      fullContent &&
+      fullContent.toLowerCase().includes(searchTerm.toLowerCase())
+    ) {
+      expect(fullContent.toLowerCase()).toContain(searchTerm.toLowerCase())
+    } else {
+      // If full content does not contain the search term, check the summary content
+      const summaryContent = await page.textContent(
+        ".Summary__TitleFormat-sc-791f19-1"
+      )
+      if (summaryContent) {
+        expect(summaryContent.toLowerCase()).toContain(searchTerm.toLowerCase())
+      } else {
+        console.warn(
+          `Both full content and summary content for the first bill on page were null or did not contain the search term.`
+        )
+      }
+    }
+
+    await page.goBack()
+    await page.waitForTimeout(1000)
```

Comment:
For context, why are these `waitForTimeout`s necessary? Is there a way to wait for the actual page state we need directly?

If possible, I'd prefer to avoid manual waits (e.g. if we wait 1 second and only need 0.1 seconds, we're slowing down the tests unnecessarily. If we wait 1 second and actually need 2 seconds, it could lead to test flappiness.)

Reply:
To address these issues, we can wait for the bills to be visible instead of using fixed timeouts.